File: bot/handlers/queue.py
import math

import discord
import lavalink
from discord.ext import commands

import config as cfg

from ..handlers import embeds
from ..handlers.voice import VoiceHandler
from ..util import models


class QueueHandler:
    def __init__(self, bot: models.LavaBot, voice_handler: VoiceHandler) -> None:
        self.bot = bot
        self.voice_handler = voice_handler

    def update_pages(self, player: lavalink.DefaultPlayer):
        player.store("pages", math.ceil(len(player.queue) / 9))

    async def clear(self, interaction: discord.Interaction, index: int = None):
        player = await self.voice_handler.ensure_voice(interaction)

        if not index:
            player.queue.clear()
            player.store("pages", 0)
            await interaction.followup.send(":boom: Queue cleared!")

        else:
            cleared = player.queue.pop(index - 1)

            if not cleared:
                await interaction.followup.send(
                    f"Failed to clear track: Track at index {index} not found. Index must be between 1 and {len(player.queue)}"
                )

            embed = embeds.ClearedEmbedBuilder(interaction, cleared, player, index)
            await interaction.followup.send(embed=embed.construct())

File: src/util/handlers/queue.ts
import { Player } from "erela.js";
import { ExtendedClient } from "../structures";
import { VoiceHandler } from "./voice";
import config from "../../config.json"
import { CommandInteraction } from "discord.js";
import { ClearedEmbedBuilder } from "../structures";

export class QueueHandler {

    private client: ExtendedClient
    private VH: VoiceHandler

    constructor(client: ExtendedClient) {
        this.client = client
        this.VH = new VoiceHandler(client)
    }

    updatePages(player: Player) {
        player.set('pages', Math.ceil(player.queue.length / config.music.listPageLength))
    }

    async clear(interaction: CommandInteraction, index?: number) {
        const player = await this.VH.ensureVoice(interaction)
        if (!player) {
            return
        }

        if (!index) {
            player.queue.clear()
            player.set('pages', 0)
            interaction.reply(":boom: Queue cleared!")

        } else {
            const cleared = player.queue.remove(index - 1).at(0)

            if (!cleared) {
                interaction.reply(config.error.trackNotFound)
                return
            }

            const embed = new ClearedEmbedBuilder(interaction, cleared, player, index).toJSON()
            await interaction.reply({embeds: [embed]})
        }
    }
}